### PR TITLE
Document supported versions of Django; fix Travis to test these versions.

### PR DESCRIPTION
**Pull request by jdufresne**
Remove workarounds for unsupported, unmaintained versions of Django.

Per the 0.21 release notes, django-taggit only supports Django 1.7 and
above.

Updated docs to only refer to supported versions.

---

I think you should also consider dropping support for Django 1.7. Is it no longer maintained or supported upstream, including for security issues. See <https://www.djangoproject.com/download/#supported-versions>. If you think it is a good idea, I can update the PR to remove all 1.7 workarounds as well.

### CODE REVIEW

File: docs/index.txt [hunk 1]
```diff
@@ -4,25 +4,11 @@ Welcome to django-taggit's documentation!
...
-
-   Since version 0.12.0 taggit uses Django migrations by default. South users
-   have to adjust their settings::
-
-      SOUTH_MIGRATION_MODULES = {
-          'taggit': 'taggit.south_migrations',
-      }
-
-   For more information, see `south documentation`__
+   Since version 0.12.0 taggit uses Django migrations by default.
```
Comment (timgraham) [inline docs/index.txt]: I think this could be removed since Django 1.6 is no longer supported.

**Comment by timgraham** [comment]
LGTM, another thing I noticed is that you can use `functools.total_ordering` in replace of the django import.

**Comment by jdufresne** [comment]
Thanks for the review! Made the recommended changes.

**Comment by frewsxcv** [comment]
Thanks for your contribution! 🎉

**Pull request merged**